apps/condo/domains/acquiring/schema/MultiPayment.js

```diff
@@ -250,27 +257,48 @@ const MultiPayment = new GQLListSchema('MultiPayment', {
                 if (alreadyWithMPPayments.length) {
                     addValidationError(`${MULTIPAYMENT_PAYMENTS_ALREADY_WITH_MP} Failed ids: ${alreadyWithMPPayments.join(', ')}`)
                 }
+
+                const receiptPayments = payments.filter(({ invoice, receipt }) => !!receipt && !invoice)
```

Comment:
```js
const { receiptPayments, invoicePayments, virtualPayments } = payments.reduce(
  (acc, { invoice, receipt }) => {
    if (!!receipt && !invoice) {
      acc.receiptPayments.push({ invoice, receipt })
    } else if (!!invoice && !receipt) {
      acc.invoicePayments.push({ invoice, receipt })
    } else if (!invoice && !receipt) {
      acc.virtualPayments.push({ invoice, receipt })
    }
    return acc
  },
  { receiptPayments: [], invoicePayments: [], virtualPayments: [] }
);
```
This code will walk through array only once

Reply:
Refactored